File: nodejs/get_phone_temperature/temp_test.py
import os, sys, sqlite3, time

print os.getcwd()
conn = sqlite3.connect("tempData.db")
query = """create table IF NOT EXISTS temperature(
    thermal VARCHAR(20),
    type VARCHAR(20),
    temperature Integer,
    time Integer   
);"""
conn.execute(query)
conn.commit()
print "create table succ!!"

process = os.popen('adb shell ls /sys/class/thermal/', "r", -1)
dirs=process.read().split("\r\n")
dirs.remove("")
print dirs


timeStamp = int(time.time())
print timeStamp
for dir in dirs:
    type = os.popen("adb shell cat /sys/class/thermal/" + dir + "/type", "r", -1).read().strip("\r\n")
    temp = os.popen("adb shell cat /sys/class/thermal/" + dir + "/temp", "r", -1).read().strip("\r\n")
    print "value:", type, " :", temp
    cmd = "insert into temperature(thermal, type, temperature, time) values (" + dir + "," + type + "," +  temp + "," + str(timeStamp) + ")"
    print cmd
    conn.execute(cmd)
    conn.commit()

    

File: nodejs/get_phone_temperature/temp_test.js
var process = require('child_process');
var SqliteDB = require('./sqlite.js').SqliteDB;

var file = "data.db";
var SqliteDB = new SqliteDB(file);


function createTable(){
    var createTableSql = "create table if not exists tempTable(name varchar(128), type varchar(128), temp float, date double)";
    SqliteDB.createTable(createTableSql);
}

function insertData(item, type, temp, date){
    var insertSql = "insert into tempTable(name, type, temp, date) values (?, ?, ?, ?)";
    SqliteDB.insertData(insertSql, [[item, type, temp, date]]);
}


function runProcess(date){
    var value = process.execSync("adb shell ls /sys/class/thermal/");
    console.log("value is " + value.toString());
    var strs = value.toString().split("\n");
    console.log("strs:" + strs[0]);
    
    strs.forEach(function(item,index){
        try{
            var type = process.execSync("adb shell cat /sys/class/thermal/" + item + "/type").toString().replace(/[\r\n]/g,"").trim();
            var temp = process.execSync("adb shell cat /sys/class/thermal/" + item + "/temp").toString().replace(/[\r\n]/g,"").trim();
            var tempInt = parseInt(temp);
            var tempReal = 0.0
            if (tempInt > 1000){
                tempReal =  tempInt / 1000;
            }else if(tempInt > 100){
                tempReal = tempInt / 10;
            }else{
                tempReal = temp;
            }
            console.log("item:" + item.toString());
            console.log("temp:" + tempReal);
            console.log("type:" + type);
            if(temp != 0){
                // write to database
                insertData(item, type, tempReal, date);
            }
        }catch(e){
            console.log("catch exception! " + item);
        }
    });
}


createTable()
setInterval(function(){
    var date = Date.now();
    runProcess(date)
}, 30 * 1000);

console.log("OVER??");
